src/screens/details.tsx

```diff
@@ -1,19 +1,209 @@
+  dest: string
+  wait: string
+  carriages: string
+  status: string
+}
+
+class StopData {
+  stopName = ''
+  incomingTrams: IncomingTram[] = []
+}
+
+export default function DetailsScreen() {
+  const [showDropDown, setShowDropDown] = useState(false)
+  const [stop, setStop] = useState('')
+  const [incomingTrams, setIncomingTrams] = useState([] as IncomingTram[])
+  const [stopsObtained, setStopsObtained] = useState(
+    [] as { label: string; value: string }[]
+  )
+
+  async function getAllStops() {
+    if (stopsObtained.length > 0) {
+      return
+    }
+
+    const stops: { label: string; value: string }[] = []
+    const res = await fetch(tfgmEndpoint, {
+      method: 'GET',
+      headers: {
+        'Ocp-Apim-Subscription-Key': config.apiKey,
+      },
+    })
+    const json = (await res.json()) as { value: APIType }
+    for (const item of json.value) {
+      const truncAtcoCode = item.AtcoCode.substring(0, item.AtcoCode.length - 1)
+      const newStop = { label: item.StationLocation, value: truncAtcoCode }
+
+      if (
+        !stops.some((value) => {
+          return value.value === truncAtcoCode
+        })
+      ) {
+        stops.push(newStop)
+      }
+    }
+
+    setStopsObtained(stops)
+  }
+
+  async function handleClick() {
+    const res = await fetch(tfgmEndpoint, {
+      method: 'GET',
+      headers: {
+        'Ocp-Apim-Subscription-Key': config.apiKey,
+      },
+    })
+    const json = (await res.json()) as { value: APIType }
+    const screenData = filterJson(json.value)
+    const stopData = pidDataToStopData(screenData)
+
+    setIncomingTrams(stopData.incomingTrams)
+  }
+
+  function pidDataToStopData(pidData: APIType): StopData {
+    const stopData = new StopData()
+    if (pidData.length === 0) {
+      return stopData
+    }
+
+    stopData.stopName = pidData[0].StationLocation
+    for (const platformData of pidData) {
+      stopData.incomingTrams.push({
+        dest: platformData.Dest0,
+        wait: platformData.Wait0,
+        status: platformData.Status0,
+        carriages: platformData.Carriages0,
+      })
+      stopData.incomingTrams.push({
+        dest: platformData.Dest1,
+        wait: platformData.Wait1,
+        status: platformData.Status1,
+        carriages: platformData.Carriages1,
+      })
+      stopData.incomingTrams.push({
+        dest: platformData.Dest2,
+        wait: platformData.Wait2,
+        status: platformData.Status2,
+        carriages: platformData.Carriages2,
+      })
+    }
+
+    stopData.incomingTrams = stopData.incomingTrams.filter(
+      (a) => !(a.wait === '')
+    )
+
+    stopData.incomingTrams.sort((a, b) => parseInt(a.wait) - parseInt(b.wait))
+
+    stopData.incomingTrams.sort((a, b) => {
+      if (a.status === 'Departing' && b.status === 'Arrived') {
+        return -1
+      }
+      if (b.status === 'Departing' && a.status == 'Arrived') {
+        return 1
+      }
+      return 0
+    })
+
+    return stopData
+  }
```

Comment:
Does this need to be in the component? It's also very long, any way to make it more redable?